crates/circuit/src/operations.rs

```diff
@@ -2860,8 +2848,21 @@ impl Operation for UnitaryGate {
         }
     }
     fn definition(&self, _params: &[Param]) -> Option<CircuitData> {
-        None
+        Python::with_gil(|py| -> Option<CircuitData> {
+            let py_op = self
+                .create_py_op(py, &ExtraInstructionAttributes::default())
+                .unwrap();
+            match py_op.getattr(py, intern!(py, "definition")) {
+                Ok(definition) => definition
+                    .getattr(py, intern!(py, "_data"))
+                    .ok()?
+                    .extract::<CircuitData>(py)
+                    .ok(),
+                Err(_) => None,
+            }
+        })
```

Comment:
Do we need a definition for `UnitaryGate` at all? I purposely didn't define it here because the python space one is to call unitary synthesis. In rust it's a bit of a layer violation to do that since it lives in the accelerate crate. While going through python avoids that issue, I'm wondering if it's really necessary because it's not a definition in the strict sense like it is for a concrete gate.